XamarinCommunityToolkit/Helpers/BindablePropertyBuilder.cs

```diff
@@ -0,0 +1,161 @@
+		public BindablePropertyBuilder SetPropertyChangingDelegate(BindableProperty.BindingPropertyChangingDelegate propertyChangingDelegate)
+		{
+			bpParameters.BindingPropertyChangingDelegate = propertyChangingDelegate;
+			return this;
+		}
+
+		/// <summary>
+		/// Sets the delegate used to coerce the range of a value.
+		/// </summary>
+		/// <param name="coerceValueDelegate">A delegate used to coerce the range of a value.</param>
+		/// <returns>The <see cref="BindablePropertyBuilder"/> to support the fluent syntax</returns>
+		public BindablePropertyBuilder SetCoerceValueDelegate(BindableProperty.CoerceValueDelegate coerceValueDelegate)
+		{
+			bpParameters.CoerceValueDelegate = coerceValueDelegate;
+			return this;
+		}
+
+		/// <summary>
+		/// Sets the delegate used to initialize the default value for reference types.
+		/// </summary>
+		/// <param name="createDefaultValueDelegate">The delegate used to initialize the default value for reference types.</param>
+		/// <returns>The <see cref="BindablePropertyBuilder"/> to support the fluent syntax</returns>
+		public BindablePropertyBuilder SetCreateDefaultValueDelegate(BindableProperty.CreateDefaultValueDelegate createDefaultValueDelegate)
+		{
+			bpParameters.CreateDefaultValueDelegate = createDefaultValueDelegate;
+			return this;
+		}
+
+		/// <summary>
+		/// Instantiates the <see cref="BindableProperty"/> based on the values that have been set.
+		/// </summary>
+		/// <returns>An instance of <see cref="BindableProperty"/> with the values that have been set</returns>
+		/// <exception cref="ArgumentNullException">
+		/// If any of the following are null:
+		/// <see cref="BindableProperty.PropertyName"/>
+		/// <see cref="BindableProperty.ReturnType"/>
+		/// <see cref="BindableProperty.DeclaringType"/>
+		/// </exception>
+		/// <exception cref="ArgumentException">
+		/// If the type of the <see cref="BindableProperty.DefaultValue"/>
+		/// and <see cref="BindableProperty.ReturnType"/> don't match.
+		/// </exception>
+		public BindableProperty Build()
+		{
+			if (string.IsNullOrEmpty(bpParameters.PropertyName))
+				throw new ArgumentNullException(nameof(bpParameters.PropertyName), $"{nameof(bpParameters.PropertyName)} is required");
+			if (bpParameters.ReturnType == default)
+				throw new ArgumentNullException(nameof(bpParameters.ReturnType), $"{nameof(bpParameters.ReturnType)} is required");
+			if (bpParameters.DeclaringType == default)
+				throw new ArgumentNullException(nameof(bpParameters.DeclaringType), $"{nameof(bpParameters.DeclaringType)} is required");
+
+			if (bpParameters.DefaultValue != null && bpParameters.ReturnType != bpParameters.DefaultValue.GetType())
+				throw new ArgumentException($"The return type is {bpParameters.ReturnType.Name}, but the default value is of type {bpParameters.DefaultValue.GetType().Name}");
+
+			var bp = BindableProperty.Create(propertyName: bpParameters.PropertyName,
```

Comment:
```return BindableProperty.Create(...``` ?

Reply:
It's a habit of mine - when I'm debugging, it's easier to see what I'm returning. I can change it if you all really want me to, it's just my preference to toss it into a variable and then return the variable.

Reply:
up to you then 👍